config/routes.rb

```diff
@@ -1,12 +1,12 @@
 Rails.application.routes.draw do
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
-  # Defines the root path route ("/")
-  # root "articles#index"
-
   root 'users#index'
 
   resources :users, only: %i[index show] do
-    resources :posts, only: %i[index show]
+    resources :posts, only: %i[index show create] do
+      resources :comments, only: :create
+      resource :like, only: :create
```

Comment:
- [ ] Dear @0sugo, According to the requirements we need to create a separate route for the `comments#new` and `comments#create` and I can see you only have a create route for your comments, Please add the new route for the comment. 
- [ ] Also, It's expected that you create a separate view for the comment in `views/comments/new.html.erb`, This makes your app more organized and visible.